File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let i = 0;
    let j = i + 1;
    let n = nums.length;
    while(j < n) {
        if(nums[i] == nums[j]) {
            j++;
        } else {
            nums[++i] = nums[j];
        }
    }

    return i+1;
};

File: 26-remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int seen;
        int i = 0;
        int start = 0;
        seen = nums[0];
        while (i < nums.length) {
            if(nums[i] != seen) {
                seen = nums[i];
                nums[++start] = nums[i];
            }
            i++;
        }
        return start+1;
    }
}